README.md

```diff
@@ -0,0 +1,211 @@
+
+The following command generates the key pair in PEM format. The key pair is protected by password of your choice. 
+
+```sh
+java -jar ssl-pinning-tool.jar keygen -o keypair.pem -p [password]
+```
+
+**Store the key pair and private key password safely! You will need it next time you replace SSL certificate to generate new signatures.**
+
+### Prepare Certificate Signature
+
+This command will retrieve the SSL certificate into PEM file:
+
+```sh
+openssl s_client -showcerts -connect my.domain.com:443 -servername my.domain.com < /dev/null | openssl x509 -outform PEM > cert.pem
+```
+
+Make sure to replace `my.domain.com` in both `-connect` and `-servername` options with your domain name.
+
+This command will generate the SSL certificate signature:
+
+```sh
+java -jar ssl-pinning-tool.jar sign -k keypair.pem -c cert.pem -o output.json -p [password]
+```
+
+The output file will contain SSL certificate signature:
+```json
+{
+  "name" : "my.domain.com",
+  "fingerprint" : "jymEKdgGPv1zSp61CYya5c2fR9fTLe8tKnWF6857iLA=",
+  "expires" : 1543322263000,
+  "signature" : "MEUCICOs9bb6TIEmRNHCekxn9URADLYuuZnk4aftpVDzdwmWAiEAlU2r9VDEnAWryxvbAsSJfIlCQjKfumdFbZeUKda166w="
+}
+``` 
+
+Alternatively, you can prepare certificate signature from information about the certificate. This approach is useful when the certificate is not deployed yet.
+
+You need following information:
+* Domain common name, e.g. `my.domain.com`
+* Certificate fingerprint in HEX format, for example: `8f298429d8063efd734a9eb5098c9ae5cd9f47d7d32def2d2a7585ebce7b88b0`
+* SSL certificate expiration time as Unix timestamp, e.g. `1543322263000`
+
+This command will generate the SSL certificate signature:
+
+```sh
+java -jar ssl-pinning-tool.jar sign -k keypair.pem -f 8f298429d8063efd734a9eb5098c9ae5cd9f47d7d32def2d2a7585ebce7b88b0 -t 1543322263000 -n my.domain.com -o output.json -p [password]
+```
+
+The output file will contain JSON with SSL certificate signature:
+```json
+{
+  "name" : "my.domain.com",
+  "fingerprint" : "jymEKdgGPv1zSp61CYya5c2fR9fTLe8tKnWF6857iLA=",
+  "expires" : 1543322263000,
+  "signature" : "MEUCICOs9bb6TIEmRNHCekxn9URADLYuuZnk4aftpVDzdwmWAiEAlU2r9VDEnAWryxvbAsSJfIlCQjKfumdFbZeUKda166w="
+}
+``` 
+
+### Export Public Key
+
+Mobile app developers will need the public key from generated key pair in order to be able to verify signatures.
+
+You can convert EC private key to public key and print it:
+
+```sh
+java -jar ssl-pinning-tool.jar export -k keypair.pem -p [password]
+```
+
+### Troubleshooting
+
+Error: 
+```
+SEVERE: Failed to load private key, error: unable to read encrypted data: javax.crypto.BadPaddingException: pad block corrupted
+```
+
+This error is shown when the private key password is invalid.
+
+Alternatively, you can perform all steps above using `openssl` command.
+
+## Prepare Data Using OpenSSL
+
+In case you have your SSL certificate in a `pem` format, you are able to generate all necessary data using OpenSSL.
```

Comment:
We should add a shell script for download of PEM version of certificate using openssl, similar to the snipped used with Java tool.

Reply:
We do an unecessary conversion of PEM to DER, the script could prepare the DER version to avoid conversion.